### ISSUE TITLE
[TW-1677] add: Race condition in the notification

### ISSUE THREAD

**Issue by taskwarrior** [GothenburgBitFactory/taskwarrior#1701]
_Daniel Shahaf on 2015-08-30T08:01:09Z says:_ 

 With the following on-add hook:

```
1. !/bin/sh
if ! [ ${TW1677:+set} ]; then
  TW1677=set task add $(date +t%s)
fi
cat
```

I get the following notifications:

```
% task add one
Created task 1.
Created task 1.
% task add two
Created task 3.
Created task 3.
% task ls  

ID Description    
 2 one
 1 t1440922196    
 3 t1440922198
 4 two            

4 tasks
```

The second notification in each pair is wrong: it specifies `1` or `3` as the id instead of `2` or `4`.

**Comment by taskwarrior** [comment]
Migrated metadata:
```
Created: 2015-08-30T08:01:09Z
Modified: 2017-01-16T18:03:48Z
```

**Comment by taskwarrior** [comment]
_Daniel Shahaf on 2015-08-30T08:21:04Z says:_ 

 To be clear, hooks are a red herring here; the issue is about concurrent invocations of `task add`.

**Comment by djmitche** [comment]
I believe this is fixed in 3.0.
